src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartBResource.java

```diff
@@ -181,4 +190,37 @@ public ResponseEntity<Void> deleteFormRPartBById(@PathVariable String id) {
       return ResponseEntity.notFound().build();
     }
   }
+
+  /**
+   * Generate a FormR PartB PDF, unless one already exists.
+   *
+   * @return The downloaded or generated PDF.
+   * @throws IOException A new PDF could not be generated.
+   */
+  @PutMapping(value = "/formr-partb-pdf", produces = MediaType.APPLICATION_PDF_VALUE)
+  public ResponseEntity<byte[]> generatePdf(@Valid @RequestBody FormRPartBDto formRPartB)
```

Comment:
I should have picked this up earlier, but do we want to trust the incoming DTO?

The alternative would be similar to [LTFT](https://github.com/Health-Education-England/tis-trainee-forms/blob/04b6589250c954b6e3dd862167e4c01fc21f7a90/src/main/java/uk/nhs/hee/tis/trainee/forms/api/LtftResource.java#L106) where we have two identical `GET` mappings that are selected based on the accepted type header.

Are we taking the DTO as parameter in this case because we may not have the latest data available to us (either the S3/DB split or because they're printing an updated copy they haven't saved)?

Reply:
(Sorry, my latest reply was without having seen your prior comment. Dratted page-refresh issue, I often have it)